File: ios/ReactNativeModules/ExampleModule.m
//
//  ExampleModule.m
//  ReactNativeHybrid
//
//  Created by Oscar Cheung on 20/02/2023.
//

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(ExampleModule, NSObject)

  RCT_EXTERN_METHOD(log:(NSString *) message)
  RCT_EXTERN__BLOCKING_SYNCHRONOUS_METHOD(randSync)
  RCT_EXTERN_METHOD(rand: (RCTPromiseResolveBlock) resolve
                    rejecter: (RCTPromiseRejectBlock) reject)
  RCT_EXTERN_METHOD(triggerEvent:(NSString *) eventName
                    params: (NSDictionary *) eventParams)

@end

File: ios/ReactNativeModules/ExampleModule.swift
//
//  ExampleModule.swift
//  ReactNativeHybrid
//
//  Created by Oscar Cheung on 20/02/2023.
//

@objc(ExampleModule)
class ExampleModule: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool { return false }
  
  @objc(log:)
  func log(_ message: String) -> Void {
    print("LoggerModule:", message)
  }

  @objc
  func randSync() -> [Float] {
    return [Float.random(in: 0...1)]
  }

  @objc
  func rand(_ resolve: RCTPromiseResolveBlock,
            rejecter reject: RCTPromiseRejectBlock) -> Void {
    do {
      try resolve(Float.random(in: 0...1))
    } catch {
      reject("0", "Error occurred", error)
    }
  }

  @objc
  func triggerEvent(_ eventName: String, eventParams: NSDictionary?) -> Void {
    
  }
  
}
